src/fluent_linter/linter.py

```diff
@@ -387,6 +389,64 @@ def visit_Identifier(self, node):
                 f"Identifiers must be at least {self.config['ID02']['min_length']} characters long",
             )
 
+        self._validate_id03(
+            node, node.name, message_id, self.state.get("is_term", False)
+        )
+
+    def visit_Junk(self, node):
+        # Try to extract and validate identifiers for ID03 checks
+        if "ID03" in self.config and self.config["ID03"]["enabled"]:
+            content = node.content
+            if "=" in content:
+                potential_id = content.split("=")[0].strip()
+                self._validate_id03(node, potential_id, potential_id)
+
+        # Add syntax error for junk content
+        self.add_error(
+            node,
+            node.content.strip() if hasattr(node, "content") else "",
+            "SY02",
+            "Syntax error in Fluent file - malformed entry detected",
+        )
+
+    def _validate_id03(self, node, identifier, message_id, is_term=False):
```

Comment:
```suggestion
    def _validate_id03(self, node, identifier, message_id):
```

You're not using it anymore.